pepicrft/catalysis-framework
# Set up Typescript, tests suite, CI, linting, and Prettier

I'm setting up a bunch of things in the repository:
- Typescript to be able to write and ship code with more confidence.
- A tests suite using [Vitest](https://vitest.dev)
- Continuous integration through [GitHub Actions](https://github.com/features/actions)
- Linting and prettier

## How to test the changes
1. Check out the branch.
2. Install dependencies with `pnpm install`
3. `cd packages/cli`.
4. Things to try:
  - `pnpm run test`: should run the tests.
  - `pnpm run build`: should build the CLI and place it under `./dist`
  - Add some code to `src/index.ts` and save. VSCode should run prettier and format automatically the code.
  - `pnpm run tc`: should run the Typescript type checker.
